### ISSUE TITLE
[BUG] Webui Behind Traefik 2.2

### ISSUE THREAD

**Issue by dragon2611** [longhorn/longhorn#1537]
**Describe the bug**
The webui seems rather buggy, I'm not sure if it's because it's being accessed behind traefik 2.2 as an ingress controller (ACL'd ofcourse, didn't leave it open to the internet)

I was manually creating some volumes using the webUI and playing with the replication settings.etc as well as attaching them to a node so I could copy some files in that the pod will need (Migrating container from a standalone docker host)

What I'm finding in both edge (chromium) and firefox is the webUI seems rather buggy in the sense you'll tell it do something that would make changes to longhorns configuration and it won't error but it won't do it either, nothing seems to happen at all.

But if you try and repeat the action it will usually work on the 2nd or 3rd attempt.

**To Reproduce**
1. Longhorn webui'
2. Click on attach volume
3. select host
4. press ok

**Expected behavior**
Longhorn attaches volume to node (or at least errors)

**Environment:**
 - Longhorn version: 1.0
 - Kubernetes version: v1.18.4+k3s1
 - Node OS type and version: Linux X86_64, Ubuntu 18.04

**Comment by dragon2611** [comment]
It looks like something to with CORS (in edge at least) after a while the websockets start failing with 

`Access to XMLHttpRequest at 'wss://lh.mydomain/v1/volumes/volname?action=pvcCreate' from origin 'https://longhorn.drag0n.ovh' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome-extension, edge, https.`

**Comment by yasker** [comment]
Sounds like the duplicate of https://github.com/longhorn/longhorn/issues/1442#issuecomment-639761799 , can you try the workaround there?

**Comment by dragon2611** [comment]
That seems to fix it once you create that middleware and add it to the routing configuration for the UI 👍
